packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -306,7 +306,18 @@ class ShortcutManager extends ChangeNotifier with Diagnosticable {
       return false;
     }
     assert(context != null);
-    final LogicalKeySet keySet = keysPressed ?? LogicalKeySet.fromSet(RawKeyboard.instance.keysPressed);
+    LogicalKeySet keySet = keysPressed;
+    if (keySet == null) {
+      if (RawKeyboard.instance.keysPressed.isNotEmpty) {
+        keySet = LogicalKeySet.fromSet(RawKeyboard.instance.keysPressed);
+      } else {
+        // Somehow we received a key down event when no keys were down.
```

Comment:
a fly by comment, shouldn't we assert in this case to indicate whoever simulating the keys event are doing it wrong?